src/Network/Request.php

```
@@ -1172,6 +1172,94 @@ public function param($name)
+	 *    which will look the exception up in namespace \Cake\Network\Exception\*Exception
+	 *    or a fully qualified namespaced exception class.
+     *
+     * Examples:
+     *
+	 * Returns '1':
+	 * $this->request->extract('Users.status', ['return' => 'value']);
+	 *
+	 * Returns ['Profiles.status' => 'bar] for query-string ?Users[Profiles][status]=bar:
+	 * $this->request->extract('Users.Profiles.status', ['return' => 'value']);
+	 *
+	 * Examples for feeding the ORM:
+	 *
+	 * $query = $this->User->find();
+	 *
+	 * Supplies the ORM's where condition with the Users.status set via query string,
+	 * defaulting to 1 in case it is missing:
+     * $query->where($this->request->extract('Users.status', ['default' => 1]));
+	 *
+     * Explicitly fetches the User status from the query string with no default and
+	 * supplies the value to the ORM's condition call:
+	 * $query->where($this->request->extract('Users.status', ['query']));
+	 *
+     * Explicitly fetches the User status from the data and
+	 * and in case it is missing throws a BadRequestException:
+     * $query->where($this->request->extract('Users.status', ['data', 'missing' => 'BadRequest']));
+     *
+     * @param string|array $path The path being searched for in form of a dot separated string.
+     * @param array $options An array of $options to configure the extraction.
+     * @return array
+     */
+	public function extract($path, $options = array('query'))
+	{
+		if (empty($options[0]) || !in_array($options[0], ['query', 'data'])){
+			$options[0] = 'query';
+		}
+		$context = $options[0];
+		$value = \Cake\Utility\Hash::get($this->{$context}, $path);
+		if (empty($value) && $value !== '0') {
+			if (isset($options['default'])) {
+				$value = $options['default'];
+			} else if (!empty($options['missing'])) {
+				if (stripos($options['missing'], '\\') === 0) {
+					throw new $options['missing'];
+				} else {
+					$exception = '\Cake\Network\Exception\\' . $options['missing'] . 'Exception';
```

At the moment you can throw any exception by supplying an absolute path starting with \.
You can even throw exceptions that are not within the network sub namespace.

So you suggest: http://book.cakephp.org/3.0/en/core-libraries/app.html#Cake\Core\App::classname
```php
throw App::classname($exception, 'Network/Exception', 'Component', 'Exception');
```
?
Would and should it be possible to throw exceptions not in \Cake|Plugin\Network\Exception NS?